test/jasmine/tests/config_test.js

```diff
@@ -470,7 +470,33 @@ describe('config argument', function() {
             var editBox = document.getElementsByClassName('plugin-editable editable')[0];
             expect(editBox).toBeUndefined();
         });
+    });
+
+    describe('plotlyServerUrl:', function() {
+        var gd;
+
+        beforeEach(function() {
+            gd = createGraphDiv();
+        });
 
+        afterEach(destroyGraphDiv);
 
+        it('should default to plotly cloud', function(done) {
+            Plotly.plot(gd, [], {})
+            .then(function() {
+                expect(gd._context.plotlyServerUrl).toBe('https://plot.ly');
+            })
+            .catch(failTest)
+            .then(done);
+        });
+
+        it('can be set to other base urls', function(done) {
+            Plotly.plot(gd, [], {}, {plotlyServerUrl: 'dummy'})
+            .then(function() {
+                expect(gd._context.plotlyServerUrl).toBe('dummy');
```

Comment:
These tests aren't great, but I couldn't find an easy way to mock the form submit in 

https://github.com/plotly/plotly.js/blob/bcab36e5c27c7227b7952b1ecc14becdd18104f0/src/plots/plots.js#L214-L245